File: articles/chemical-revolution-rise-modern-chemistry/c/oxidation_mass_gain.c
#include <stdio.h>

// Compact oxidation mass-gain kernel.

int main(void) {
    double metal_mass_g = 24.305;
    double oxygen_mass_g = 16.000;
    double oxide_mass_g = metal_mass_g + oxygen_mass_g;
    double oxygen_mass_fraction = oxygen_mass_g / oxide_mass_g;

    printf("oxide_mass_g=%.6f\n", oxide_mass_g);
    printf("oxygen_mass_fraction=%.6f\n", oxygen_mass_fraction);

    return 0;
}

File: articles/chemical-revolution-rise-modern-chemistry/r/oxidation_mass_gain.R
# Oxidation mass gain table using synthetic examples.

oxidation <- read.csv(file.path("data", "oxidation_mass_gain.csv"))

oxidation$oxide_mass_g <- oxidation$metal_mass_g + oxidation$oxygen_mass_g
oxidation$oxygen_mass_fraction <- oxidation$oxygen_mass_g / oxidation$oxide_mass_g
oxidation$mass_gain_percent <- 100 * oxidation$oxygen_mass_g / oxidation$metal_mass_g

print(round(oxidation, 4))
